starlite/middleware/session/base.py

```diff
@@ -0,0 +1,313 @@
+            )
+
+    async def load_from_connection(self, connection: ASGIConnection) -> Dict[str, Any]:
+        """Load session data from a connection and return it as a dictionary to
+        be used in the current application scope.
+
+        The session-ID will be gathered from a cookie with the key set in the
+        [configuration[BaseBackendConfig.key]. If a cookie is found, its value will
+        be used as the session-ID and data associated with this ID will be loaded
+        using [get][starlite.middleware.session.base.ServerSideBackend.get]. If no cookie was found or no data was loaded
+        from the store, this will return an empty dictionary.
+
+        Args:
+            connection: Originating ASGIConnection
+        """
+        session_id = connection.cookies.get(self.config.key)
+        if session_id:
+            data = await self.get(session_id)
+            if isinstance(data, dict):
+                return data
+            if data is not None:
+                return self.deserialise_data(data)
+        return {}
+
+
+B = TypeVar("B", bound=SessionBackend)
+
+
+class SessionMiddleware(MiddlewareProtocol, Generic[B]):
+    __slots__ = ("backend",)
+
+    def __init__(self, app: "ASGIApp", backend: B) -> None:
+        """Starlite session middleware for storing session data.
+
+        Args:
+            app: An ASGI application
+            backend: A [SessionBackend][starlite.middleware.session.base.SessionBackend]
+                instance used to store and retrieve session data
+        """
+
+        self.app = app
+        self.backend = backend
+
+    def create_send_wrapper(self, connection: ASGIConnection) -> Callable[["Message"], Awaitable[None]]:
+        """
+        Creates a wrapper for the ASGI send function, which handles setting the cookies on the outgoing response.
+        Args:
+            connection: ASGIConnection
+
+        Returns:
+            None.
+        """
+
+        async def wrapped_send(message: "Message") -> None:
+            """A wrapper around the send function, declared in local scope to
+            use closure values.
+
+            Args:
+                message: An ASGI message.
+
+            Returns:
+                None
+            """
+            if message["type"] != "http.response.start":
+                await connection.send(message)
+                return
+
+            scope_session = connection.scope.get("session")
+
+            await self.backend.store_in_message(scope_session, message, connection)
+            await connection.send(message)
+
+        return wrapped_send
+
+    async def __call__(self, scope: "Scope", receive: "Receive", send: "Send") -> None:
+        """
+        Args:
+            scope: The ASGI connection scope.
+            receive: The ASGI receive function.
+            send: The ASGI send function.
+
+        Returns:
+            None
+        """
+        if scope["type"] in self.backend.config.scopes:
```

Comment:
we should probably support the same kind of logic we support in the `AbstractAuthenticationMiddleware` - allowing to skip routes by either regex (`exclude`) or by using a specific `opt` key.

Reply:
Good idea, will add!

Reply:
please make sure to update the config to support this as well.

Reply:
It's done